src/libNode/FinalBlockProcessing.cpp

```diff
@@ -1384,8 +1465,9 @@ bool Node::ProcessPendingTxn(const bytes& message, unsigned int cur_offset,
   const auto& currentEpochNum =
       m_mediator.m_txBlockChain.GetLastBlock().GetHeader().GetBlockNum();
 
-  if (currentEpochNum > epochNum) {
-    LOG_GENERAL(WARNING, "PENDINGTXN sent of an older epoch " << epochNum);
+  if (currentEpochNum > epochNum + 1) {
```

Comment:
Can I ask why was this change needed? allowing one epoch later pending txn

Reply:
Because now the pending txn will be forwarded upon the completion of MicroBlockConsensus by the MBnForwardTxn message, the lookup may receive it later than the FinalBlock message due to lagging network. Besides, there is a checker for existing transactions, it can avoid duplication that if the transaction has already been confirmed but the pending transaction tries to mislead it.